File: hw4/france.py
import random

r = "republic"
m = "monarchy"
f = "facist dictatorship"
em = "Emmanuel Macron"
mlp = "Marie la Pen"
lxxb = "Louis XX Bourbon"

class Country:
    def __init__(self, continent : str, GDP : int):
        self._continent = continent
        self._GDP = GDP

    @property
    def continent(self)
        return self._continent
    
    @property
    def GDP(self)
        return self._GDP

class France(Country):

    def __init__(self, continent : str, GDP : int,govtype : str, head : str, population: int,year: int) -> None:
        Country.__init__(self, "Europe", 18999999999)
        self._govtype : str = govtype
        self._head : str = head
        self._population : str = population
        self._year : int = year
    
    def newYear(self) -> None:
        self._year = self._year + 1
        self._population = self._population*1.14

    @property
    def govtype(self):
        return self._govtype
        
    @property
    def head(self):
        return self._head

    @head.setter
    def head(self,value : str) -> None:
        self._head = value

    @property
    def year(self):
        return self._year

    @property
    def population(self):
        return self._population

File: hw5/src/france.cpp
#include "France.h"

#include <iostream>

namespace france
{
    France::France()
    : Country(13000000)
    {
        m_govtype = "Facist Dictatorship";
        m_head = "Marie LaPen";
        m_population = 60000000;
        m_year = 2030;
        std::cout << "France@" << (void *)this << " constructed." << std::endl;
    }

    France::France(std::string continent, float GDP, std::string govtype, std::string head, int population, int year)
    :Country(GDP, continent)
    {
        m_govtype = govtype;
        m_head = head;
        m_population = population;
        m_year = year;
        std::cout << "France@" << (void *)this << " constructed." << std::endl;
    }

    France::~France()
    {
        std::cout << "France@" << (void *)this << " destructed." << std::endl;
    }

    void France::newYear()
    {
        m_year++;
        m_population *= 1.14;
    }
    //getters
    std::string France::govType() const
    {
        return m_govtype;
    }
    std::string France::head() const
    {
        return m_head;
    }
    int France::population() const
    {
        return m_population;
    }
    int France::year() const
    {
        return m_year;
    }

    //setters
    void France::govType(std::string value)
    {
        m_govtype = value;
    }
    void France::head(std::string value)
    {
        m_head = value;
    }
    void France::population(int value)
    {
        m_population = value;
    }
    void France::year(int value)
    {
        m_year = value;
    }
}